tests/src/Kernel/BundlePluginTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\Tests\entity\Kernel;
+
+use Drupal\entity_module_bundle_plugin_test\Entity\EntityTestBundlePlugin;
+use Drupal\KernelTests\KernelTestBase;
+
+/**
+ * Tests the bundle plugin API.
+ *
+ * @group entity
+ */
+class BundlePluginTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'system',
+    'entity',
+    'entity_module_bundle_plugin_test',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->installSchema('system', 'router');
+    $this->installEntitySchema('entity_test_bundle_plugin');
+    $this->installConfig('entity_module_bundle_plugin_test');
+  }
+
+  /**
+   * Tests the bundle plugins.
+   */
+  public function testPluginBundles() {
```

Comment:
Can we add some EFQ asserts here as well? Just to make sure everything is dandy.